stl/inc/xmemory

```diff
@@ -1031,11 +1031,25 @@ template <class _Alloc>
 _CONSTEXPR20_DYNALLOC void _Deallocate_plain(_Alloc& _Al, typename _Alloc::value_type* const _Ptr) noexcept {
     // deallocate a plain pointer using an allocator
     using _Alloc_traits = allocator_traits<_Alloc>;
-    if constexpr (is_same_v<_Alloc_ptr_t<_Alloc>, typename _Alloc::value_type*>) {
-        _Alloc_traits::deallocate(_Al, _Ptr, 1);
-    } else {
-        using _Ptr_traits = pointer_traits<_Alloc_ptr_t<_Alloc>>;
-        _Alloc_traits::deallocate(_Al, _Ptr_traits::pointer_to(*_Ptr), 1);
+#ifdef __clang__ // TRANSITION LLVM-XXXX

```

Comment:
This will incorrectly ignore user specializations of `allocator_traits`. I think the proper fix is to avoid the optimization in `allocator_traits::deallocate` when `is_constant_evaluated()`.

Reply:
Moved to `_Default_allocator_traits::deallocate`